app/services/foreman_bootdisk/iso_generator.rb

```diff
@@ -76,12 +99,40 @@ def self.generate(opts = {})
 
         if opts[:ipxe]
           ipxe_source_file = File.join(Setting[:bootdisk_ipxe_dir], 'ipxe.lkrn')
-          raise(Foreman::Exception, N_('Please ensure the ipxe-bootimgs package is installed.')) unless File.exist?(ipxe_source_file)
+          raise Foreman::Exception.new(N_('Please ensure the ipxe-bootimgs package is installed.')) unless File.exist?(ipxe_source_file)
 
           FileUtils.cp(ipxe_source_file, File.join(wd, 'build', 'ipxe'))
           File.open(File.join(wd, 'build', 'script'), 'w') { |file| file.write(opts[:ipxe]) }
         end
 
+        if opts[:grub]
+          # a copy when using CDROM directly
+          File.open(File.join(wd, 'build', 'grub.cfg'), 'w') { |file| file.write(opts[:grub]) }
```

Comment:
When booting in UEFI, grub looks for `/efi/boot/grub.cfg` with `/` being the device it booted from. When I attached the bootdisk image as a cdrom, I got into grub prompt.
```suggestion
          FileUtils.mkdir_p(File.join(wd, 'build', 'EFI', 'BOOT'))
          File.open(File.join(wd, 'build', 'EFI', 'BOOT', 'grub.cfg'), 'w') { |file| file.write(opts[:grub]) }
```

Reply:
Yeah I have found this as well, fixed.